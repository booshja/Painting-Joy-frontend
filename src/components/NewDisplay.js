// dependencies
import React, { useEffect, useState } from "react";
import styled from "styled-components";

const StyledDiv = styled.div`
    width: 300px;
    display: flex;
    justify-content: center;
    background-color: rgba(0, 0, 0, 0.6);
    box-shadow: 4px 4px 4px rgba(0, 0, 0, 0.5);
    position: relative;
`;

const StyledImg = styled.img`
    height: 200px;
`;

const StyledTextBox = styled.div`
    background-color: rgba(0, 0, 0, 0.6);
    position: absolute;
    padding: 0.5rem;
    bottom: 0;
    right: 0;
`;

const StyledText = styled.p`
    color: #ffffff;
    font-family: "News Cycle", sans-serif;
    font-size: 1.4rem;
    letter-spacing: 1px;
`;

const NewDisplay = ({ variant, object }) => {
    return (
        <StyledDiv>
            <StyledImg src={object.image} alt={object.title} />
            <StyledTextBox>
                <StyledText>
                    {variant == "mural"
                        ? "See my latest mural!"
                        : "See what's new in the store!"}
                </StyledText>
            </StyledTextBox>
        </StyledDiv>
    );
};

export default NewDisplay;
