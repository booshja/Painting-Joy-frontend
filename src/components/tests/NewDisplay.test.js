import { render } from "@testing-library/react";
import NewDisplay from "../NewDisplay";
import "jest-styled-components";

/** Smoke Test */
it("renders without crashing, variant null", () => {
    render(
        <NewDisplay
            variant={null}
            title="This is a title"
            image="http://lorempixel.com/200/200"
            id={0}
        />
    );
});

it("renders without crashing, variant mural", () => {
    render(
        <NewDisplay
            variant="mural"
            title="This is a title"
            image="http://lorempixel.com/200/200"
            id={0}
        />
    );
});

/** Snapshot Test */
it("matches snapshot, variant null", () => {
    const { asFragment } = render(
        <NewDisplay
            variant={null}
            title="This is a title"
            image="http://lorempixel.com/200/200"
            id={0}
        />
    );
    expect(asFragment()).toMatchSnapshot();
});

it("matches snapshot, variant mural", () => {
    const { asFragment } = render(
        <NewDisplay
            variant="mural"
            title="This is a title"
            image="http://lorempixel.com/200/200"
            id={0}
        />
    );
    expect(asFragment()).toMatchSnapshot();
});

/** RTL Tests */
it("renders correctly, variant null", () => {
    const { getByAltText, getByText } = render(
        <NewDisplay
            variant={null}
            title="This is a title"
            image="http://lorempixel.com/200/200"
            id={0}
        />
    );

    expect(getByAltText("This is a title")).toBeInTheDocument();
    expect(getByText("See what's new in the store!")).toBeInTheDocument();
});

it("renders correctly, variant mural", () => {
    const { getByAltText, getByText } = render(
        <NewDisplay
            variant="mural"
            title="This is a title"
            image="http://lorempixel.com/200/200"
            id={0}
        />
    );

    expect(getByAltText("This is a title")).toBeInTheDocument();
    expect(getByText("See my latest mural!")).toBeInTheDocument();
});
