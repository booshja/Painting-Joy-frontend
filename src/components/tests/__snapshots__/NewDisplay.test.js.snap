// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-bdnxRM bEPPyi"
  >
    <img
      class="sc-gtsrHT hQtePG"
    />
    <div
      class="sc-dlnjwi jTWZkz"
    >
      <p
        class="sc-hKFxyN fjSfXV"
      >
        See what's new in the store!
      </p>
    </div>
  </div>
</DocumentFragment>
`;
